src/neo/Cryptography/MPT/MPTTrie.cs

```diff
@@ -9,22 +10,25 @@ public partial class MPTTrie<TKey, TValue>
         where TValue : class, ISerializable, new()
     {
         private const byte Prefix = 0xf0;
-
+        private readonly bool full;
         private readonly ISnapshot store;
+        private readonly DataCache<UInt256, ByteArrayWrapper> cache;
```

Comment:
Is this necessary? I think leveldb or other database has cache too.

Reply:
I will look into it.

Reply:
I use this code to test get one data from leveldb snapshot and dictionary.
Output:
 `00:00:00.0287010`
 `00:00:00.0051700`
```
            using var store = new LevelDBStore().GetStore();
            using var snapshot = store.GetSnapshot();
            snapshot.Put(0x00, keyBytes, val);
            var t = DateTime.UtcNow;
            for (int i = 0; i < 100000; i++)
            {
                snapshot.TryGet(0x00, keyBytes);
            }
            Console.WriteLine(DateTime.UtcNow - t);

            var dict = new Dictionary<UInt256, byte[]>();
            dict.Add(key, val);
            t = DateTime.UtcNow;
            for (int i = 0; i < 100000; i++)
            {
                dict.TryGetValue(key, out byte[] value);
            }
            Console.WriteLine(DateTime.UtcNow - t);
```
We can say it will cost more if we read data repeatedly from leveldb snapshot.

Every time we delete a node in mpt, We will `Resolve` hash again, this means we will  read data from snapshot definately again. I think we need cache.
The reason why we need read again is to make sure the `Reference` is latest. Because if the same node hash is referred by another branch it can be changed there. Besides different branch share same node data but not node instance.
we should avoid changing other branch when one branch change the node.

Resolve when modi node:
https://github.com/KickSeason/neo/blob/3a0bd310fefe35d73ce9212d9fb8e57c7e359bec/src/neo/Cryptography/MPT/MPTTrie.cs#L59
https://github.com/KickSeason/neo/blob/3a0bd310fefe35d73ce9212d9fb8e57c7e359bec/src/neo/Cryptography/MPT/MPTTrie.cs#L72

Reply:
What's the difference between `00:00:00.0287010` and `00:00:00.0051700` if we can read it 100000 times?

Reply:
I will remove cache.

Reply:
I will remove the cache. We can decide whether to optimize after we have systematic performance test results.

Reply:
Done

Reply:
@erikzhang  Just thought of a reason why I use cache is that `Get` method in all `ISnapshot` implementations reads   directly from `db` and not check `batch`. So if one wants to get key-value from snapshot just stored, he will get `null` or the old data.

Reply:
@erikzhang Erik?

Reply:
I think we should use `DataCache` with `TKey` and `TValue` directly, instead of creating a `ByteArrayWrapper`. Because `TKey` and `TValue` are already `ISerializable`.

Reply:
We don't store `TKey` and `TValue` directly. Instead we store different types `MPTNode` in database. 
I think you mean make `MPTNode` serializable.

Reply:
Done.